specification/signalr/resource-manager/Microsoft.SignalRService/preview/2024-01-01-preview/signalr.json

```diff
@@ -1597,6 +1597,180 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.SignalRService/signalR/{resourceName}/replicas/{replicaName}/sharedPrivateLinkResources": {
+      "get": {
+        "tags": [
+          "SignalR"
+        ],
+        "description": "List shared private link resources",
+        "operationId": "SignalRReplicaSharedPrivateLinkResources_List",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceName"
+          },
+          {
+            "$ref": "#/parameters/ReplicaName"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success. The response describes a list of shared private link resources.",
+            "schema": {
+              "$ref": "#/definitions/SharedPrivateLinkResourceList"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "SignalRReplicaSharedPrivateLinkResources_List": {
+            "$ref": "./examples/SignalRReplicaSharedPrivateLinkResources_List.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.SignalRService/signalR/{resourceName}/replicas/{replicaName}/sharedPrivateLinkResources/{sharedPrivateLinkResourceName}": {
```

Comment:
`delete` operation should also be supported:
https://github.com/Azure/azure-openapi-validator/blob/main/docs/all-proxy-resources-should-have-delete.md

Reply:
Delete is not supported by design. I can share some background:

A `signalr/replicas` is a replication of the primary resource `signalr` in a different region.

A `signalr/replicas/sharedPrivateLinkResources` is a replication of `signalr/sharedPrivateLinkResources`. When a `signalr/sharedPrivateLinkResources` is created, if `signalr/replicas` exists, the `signalr/sharedPrivateLinkResources` is auto replicated to each replicas, represented by `signalr/replicas/sharedPrivateLinkResources`. Customers can use `signalr/replicas/sharedPrivateLinkResources` to check status in each replica.

A `signalr/replicas/sharedPrivateLinkResources` is always created by system auto replication. Customers cannot directly create it on `signalr/replicas`. The `PUT` API is added for letting customers change some metadata. In other words, only **update** is allowed. Attempt to **create** is prohibited in RP logic. 

**Delete** of `signalr/replicas/sharedPrivateLinkResources` is supposed to be triggered by delete of `signalr/sharedPrivateLinkResources`. Customers cannot directly delete `signalr/replicas/sharedPrivateLinkResources`.

Let me know your thoughts.